### PR TITLE
Internal: block storage volume attach cleanup

### PR DESCRIPTION
**Pull request by jtopjian**
For #456

### CODE REVIEW

File: openstack/blockstorage_volume_attach_v2.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	default:
+		attachError = fmt.Errorf("Invalid attach_mode specified")
+	}
+
+	return attachMode, attachError
+}
+
+func blockStorageVolumeAttachV2ParseID(id string) (string, string, error) {
+	parts := strings.Split(id, "/")
+	if len(parts) < 2 {
```
Comment (ozerovandrei) [inline openstack/blockstorage_volume_attach_v2.go]: Maybe it's better to check for just a specific length?
I think that if length is > 2 is bad too, right?
So, what about `if len(parts) != 2 {return "", fmt.Errorf(...)}`.

File: openstack/blockstorage_volume_attach_v3.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	default:
+		attachError = fmt.Errorf("Invalid attach_mode specified")
+	}
+
+	return attachMode, attachError
+}
+
+func blockStorageVolumeAttachV3ParseID(id string) (string, string, error) {
+	parts := strings.Split(id, "/")
+	if len(parts) < 2 {
```
Comment (ozerovandrei) [inline openstack/blockstorage_volume_attach_v3.go]: Same here.

File: openstack/blockstorage_volume_attach_v2.go (hunk 1 above)
Comment (jtopjian) [reply openstack/blockstorage_volume_attach_v2.go]: Agreed. I was about to do this but wanted to be strict with not introducing any new validation to these cleanup PRs in order to limit the amount of changes.

This is a good idea to do - maybe as a separate PR. I've made a note to myself about it.

Comment (ozerovandrei) [reply openstack/blockstorage_volume_attach_v2.go]: Alright, sounds great!

**Comment by jtopjian** [comment]
Thanks, @ozerovandrei!

**Pull request merged**